File: java/RockPaperScissors.java
// return winner of a rock paper scissor game

public class RockPaperScissors {

  private static boolean rpsResult(String playerA, String playerB) {
    if (playerA == "rock" && playerB == "scissors") {
      return true;
    } else if (playerA == "scissors" && playerB == "paper") {
      return true;
    } else if (playerA == "paper" && playerB == "rock") {
      return true;
    } else
      return false;
  }

  public static String rps(String p1, String p2) {
    if (p1 == p2) {
      return "Draw!";
    } else {
      return rpsResult(p1, p2) ? "Player 1 won!" : "Player 2 won!";
    }
  }

  public static void main(String[] args) {
    System.out.println(rps("paper", "paper")); // Draw
    System.out.println(rps("rock", "paper")); // Player 2 wins
    System.out.println(rps("paper", "scissors")); // Player 2 wins
    System.out.println(rps("rock", "scissors")); // Player 1 wins
  }
}

/*
 * Let's play! You have to return which player won! In case of a draw return
 * Draw!.
 * 
 * Examples(Input1, Input2 --> Output):
 */

File: c#/RockPaperScissors.cs
// return winner of rock paper scissor

public class RockPaperScissors
{
	public static string Rps(string p1, string p2)
	{
		if (p1 == p2) return "Draw!";
		else if (
			(p1 == "rock" && p2 == "scissors")
			|| (p1 == "scissors" && p2 == "paper")
			|| (p1 == "paper" && p2 == "rock")
		) return "Player 1 won!";
		else return "Player 2 won!";
	}

	public static void Main(string[] args)
	{
		System.Console.WriteLine(Rps("paper", "rock")); // "Player 1 won!"
		System.Console.WriteLine(Rps("scissors", "rock")); // "Player 2 won!"
		System.Console.WriteLine(Rps("rock", "rock")); // "Draw!"
	}
}

// awesome solution:
//     return p1 == p2 ? "Draw!" : $"Player {(new[] {3, -1, -2}.Contains(p1[0] - p2[0]) ? 1 : 2)} won!";